src/Native/Unix/System.Native/pal_termios.cpp

```diff
@@ -0,0 +1,360 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_types.h"
+#include <termios.h>
+#include <sys/ioctl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+/* This is dup of System/IO/Ports/NativeMethods.cs */
+enum
+{
+    ParityNone = 0,
+    ParityOdd  = 1,
+    ParityEven = 2,
+    ParityMark = 3,
+    ParitySpace = 4
+};
+
+/* System/IO/Ports/Handshake.cs */
+enum
+{
+    HandshakeNone,          // No Flow Control
+    HandshakeSoft,          // Software Flow Control -> XOn/Xoff
+    HandshakeHard,          // Hardware Flow Control -> RTS/CTS
+    HandshakeBoth           // Software & Hardware Flow Control
+};
+
+static int SystemNative_TermiosGetStatus(int fd)
+{
+    int status = 0;
+    if(ioctl(fd, TIOCMGET, &status) < 0)
+    {
+        return -1;
+    }
+
+    return status;
+}
+
+static int SystemNative_TermiosGetSignal(int fd, int bit)
+{
+    int status = 0;
+    if(ioctl(fd, TIOCMGET, &status) < 0)
+    {
+        return -1;
+    }
+
+    return (status & bit) ? 1 : 0;
+}
+
+extern "C" int32_t SystemNative_TermiosGetDcd(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_CAR);
+}
+
+extern "C" int32_t SystemNative_TermiosGetCts(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_CTS);
+}
+
+extern "C" int32_t SystemNative_TermiosGetRts(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_RTS);
+}
+
+extern "C" int32_t SystemNative_TermiosGetDsr(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_DSR);
+}
+
+extern "C" int32_t SystemNative_TermiosGetDtr(int fd)
+{
+    return SystemNative_TermiosGetSignal(fd, TIOCM_DTR);
+}
```

Comment:
Do we need to expose so many externs?  Or could we just expose TermiosGetSignal as well as a PAL'd set of symbols for the bit argument?